src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,97 @@
+    private static final User USER_WITH_NULL_AGE = new User("BobAlison", "password", null);
+    private static final User USER_WITH_NULL_PASSWORD = new User("BobAlison", null, 25);
+    private static final User USER_WITH_NULL_LOGIN = new User(null, "password", 25);
+    private static final User USER_WITH_SHORT_LOGIN = new User("abc", "password", 25);
+    private static final User USER_WITH_SHORT_PASSWORD = new User("BobAlison", "abc", 25);
+    private static final User USER_BELOW_MIN_AGE = new User("BobAlison", "password", 16);
+    private static final User USER_WITH_NEGATIVE_AGE = new User("NickAlison", "password", -16);
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    public void setup() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void testRegister_ValidUser_ReturnsRegisteredUser() {
+        User registeredUser = registrationService.register(VALID_USER);
+        assertNotNull(registeredUser);
+        assertEquals(VALID_USER, registeredUser);
+    }
+
+    @Test
+    public void testRegister_NullUser_ThrowsUserIsNullException() {
+        assertThrows(UserIsNullException.class, () -> registrationService.register(NULL_USER));
+    }
+
+    @Test
+    public void testRegister_NullAge_ThrowsNotEnoughAgeException() {
+        assertThrows(NotEnoughAgeException.class, () -> registrationService.register(
+                USER_WITH_NULL_AGE));
+    }
+
+    @Test
+    public void testRegister_NegativeAge_ThrowsNotEnoughAgeException() {
+        assertThrows(NotEnoughAgeException.class, () -> registrationService.register(
+                USER_WITH_NEGATIVE_AGE));
+    }
+
+    @Test
+    public void testRegister_NullPassword_ThrowsNotEnoughSizeException() {
+        assertThrows(NotEnoughSizeException.class, () -> registrationService.register(
+                USER_WITH_NULL_PASSWORD));
+    }
+
+    @Test
+    public void testRegister_NullLogin_ThrowsNotEnoughSizeException() {
+        assertThrows(NotEnoughSizeException.class, () -> registrationService.register(
+                USER_WITH_NULL_LOGIN));
+    }
+
+    @Test
+    public void testRegister_LoginTooShort_ThrowsNotEnoughSizeException() {
+        assertThrows(NotEnoughSizeException.class, () -> registrationService.register(
+                USER_WITH_SHORT_LOGIN));
+    }
+
+    @Test
+    public void testRegister_PasswordTooShort_ThrowsNotEnoughSizeException() {
+        assertThrows(NotEnoughSizeException.class, () -> registrationService.register(
+                USER_WITH_SHORT_PASSWORD));
+    }
+
+    @Test
+    public void testRegister_AgeBelowMinimum_ThrowsNotEnoughAgeException() {
+        assertThrows(NotEnoughAgeException.class, () -> registrationService.register(
+                USER_BELOW_MIN_AGE));
+    }
+
+    @Test
+    public void testRegister_ExistingLogin_ReturnsNull() {
+        registrationService.register(EXISTING_USER);
```

Comment:
use method `add()` from StorageDaoImpl to insert some mock test data. [Try to avoid these common mistakes, while solving task](https://mate-academy.github.io/jv-program-common-mistakes/java-core/junit/registration-validation)

![image](https://github.com/mate-academy/jv-registration-validation/assets/113326347/3a2eae70-0cc6-4fe8-945a-2f617c8723c0)